src/NetscriptFunctions/Corporation.ts

```diff
@@ -431,6 +435,18 @@ export function NetscriptCorporation(
       const material = getMaterial(divisionName, cityName, materialName);
       BuyMaterial(material, amt);
     },
+    bulkPurchase: function (adivisionName: any, acityName: any, amaterialName: any, aamt: any): void {
+      checkAccess("bulkPurchase", 7);
+      const divisionName = helper.string("bulkPurchase", "divisionName", adivisionName);
+      if (!hasResearched(getDivision(adivisionName), "Bulk Purchasing")) throw new Error(`You have not researched Bulk Purchase in ${adivisionName}`)
```

Comment:
${adivisionName}

Should this be using ${divisionName} instead?

Reply:
good spot